### ISSUE TITLE
Latest: by SemVer or by timestamp?

### ISSUE THREAD

**Issue by aguschin** [iterative/gto#132]
`gto which rf prod` should show the latest version of `rf` promoted to `prod`. There are two approaches on selecting the latest one: by SemVer and by timestamp.

E.g. if I promoted `v1.0.0` first, and then promoted `v0.0.1`: latest by SemVer would be `v1.0.0`, while latest by timestamp would be `v0.0.1`.

Promoting non-registered versions makes things more interesting. E.g. if I promoted version from commit `abc1234` and didn't register a version from that commit - the latest by timestamp would be `abc1234` now.

The question is: what should we show by default in Studio and in CLI?

Since we treat SemVer as a number that positively correlates with model quality and stability, the default latest in my mind should be `v1.0.0`. But then if you have a single registered version of model in stage `prod` and then for some reason abandoned registering SemVer versions and just promote, you'll always have that SemVer in `gto latest` and `gto show`. As a more simple example, a user may be confused after promoting `abc1234` because it doesn't appear in `gto show` by default or because `gto latest` doesn't return it.

As an option to resolve this we can keep `--sort-by-semver` default and add flag `--sort-by-time` that will give the recent promoted version/ref.

WDYT?  cc @omesser @dberenbaum @dmpetrov @shcheklein

**Comment by aguschin** [comment]
@omesser, getting back to GTO since MLEM release is on the way 😅 

What you say makes sense to me, and I would like to agree considering the simplicity. But with each "strict" option we need to understand how GTO behaves if a user breaks it. If we don't allow a user to promote a model in a commit without registering it (explicitly in your option1 / implicitly in option2), then what GTO should do if a user creates that git tag by himself? I see these options:

1. Ignore - pretend that promotion didn't happen. Don't show the promotion anywhere. Have a `$ gto doctor` cmd (or `$ gto check-registry-correctness` or anything else) that will let user know that he needs to register a semver.
2. Issue warnings each time user calls some command. (Hey, you have a promotion, but don't have a version, go and register it!)
3. Break each time user calls some command. Ask to register the version to fix this.
4. Automatically register version.

I think 1st and 2nd are preferable. But still, they could be bad since promoting git tag will trigger CI if you push it. So in fact, a user may be confused: I created a tag (though manually), I pushed it, why am I getting some errors?

Also, another case to consider. Let's suppose I registered version and made a promotion. I've pushed promoting tag, but forgot to push registering tag. Then the other person clones the repo, and he's getting these warnings. 🤔 Not sure what we can do to fix this, but looks confusing. Maybe have some GTO cmd to push git tags and check that everything is ok like @dberenbaum and you suggested, e.g. [here](https://github.com/iterative/gto/issues/115#issuecomment-1103861095) (can't find the other one by Dave).
